File: BOJ/2217.cpp
#include<cstdio>
#include<algorithm>
using namespace std;

int rope[100000];

bool comp(int a, int b) {
	return a > b;
}

int main() {
	int n;
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		scanf("%d", &rope[i]);
	}
	sort(rope, rope + n, comp);

	for (int i = 1; i < n; i++) {
		rope[i] = max(rope[i - 1], (i + 1)*rope[i]);
	}
	printf("%d\n", rope[n - 1]);

	return 0;
}

File: BOJ/python/2217.py
import sys

input = sys.stdin.readline


def main() -> None:
    N = int(input().rstrip('\n'))
    rope = [int(input().rstrip('\n')) for _ in range(N)]
    rope.sort(reverse=True)

    weights = [0] * N
    for i in range(N):
        weights[i] = rope[i] * (i + 1)

    print(max(weights))


if __name__ == "__main__":
    main()
